lib/services/nativescript-project-plugins-service.ts

```diff
@@ -0,0 +1,472 @@
+			let packageJsonContent = this.getProjectPackageJsonContent().wait();
+			let pluginBasicInfo = this.getPluginBasicInformation(pluginName);
+
+			if(packageJsonContent.dependencies[pluginBasicInfo.name]) {
+				delete packageJsonContent.dependencies[pluginBasicInfo.name];
+				this.$fs.writeJson(pathToPackageJson, packageJsonContent).wait();
+				this.$logger.printMarkdown(`Successfully removed plugin \`${pluginName}\``);
+			} else {
+				this.$logger.printMarkdown(`Plugin \`${pluginName}\` is not installed.`);
+			}
+		}).future<void>()();
+	}
+
+	public configurePlugin(pluginName: string, version?: string, configurations?: string[]): IFuture<void> {
+		return (() => {
+			throw new Error("This operation is not supported for NativeScript Plugins.");
+		}).future<void>()();
+	}
+
+	public isPluginInstalled(pluginName: string): boolean {
+		let packageJsonContent = this.getProjectPackageJsonContent().wait();
+		let pluginBasicInfo = this.getPluginBasicInformation(pluginName);
+		return packageJsonContent
+				&& !!packageJsonContent.dependencies && !!packageJsonContent.dependencies[pluginBasicInfo.name]
+				&& (!pluginBasicInfo.version || packageJsonContent.dependencies[pluginBasicInfo.name] === pluginBasicInfo.version);
+	}
+
+	public getPluginBasicInformation(pluginName: string): IBasicPluginInformation {
+		let [ name, version ] = pluginName.split("@");
+		return { name, version };
+	}
+
+	public fetch(pluginIdentifiers: string[]): IFuture<void> {
+		return (() => {
+			_.each(pluginIdentifiers, pluginId => {
+				try {
+					let pathToNodeModules = path.join(this.$project.getProjectDir().wait(), NativeScriptProjectPluginsService.NODE_MODULES_DIR_NAME);
+					this.$fs.ensureDirectoryExists(pathToNodeModules).wait();
+					let currentNodeModules = this.$fs.readDirectory(pathToNodeModules).wait();
+					this.$childProcess.exec("npm install " +  pluginId).wait();
+					let newNodeModules = this.$fs.readDirectory(pathToNodeModules).wait();
+					let installedModule = _.first(_.difference(newNodeModules, currentNodeModules));
+					let pluginsPath = path.join(this.$project.getProjectDir().wait(), "plugins");
+					shelljs.mv([path.join(pathToNodeModules, installedModule)], pluginsPath);
+					this.installLocalPlugin(path.join(pluginsPath, installedModule)).wait();
+				} catch (err) {
+					this.$logger.trace(err);
+					let errorMessage =`Unable to add plugin ${pluginId}.`;
+					if(err.code === "EPERM") {
+						errorMessage += ` Make sure the ${NativeScriptProjectPluginsService.NODE_MODULES_DIR_NAME} directory of your project is not used by any other program.`;
+					} else {
+						errorMessage += " Make sure this is a valid plugin name, path to existing directory or github URL.";
+					}
+					this.$errors.failWithoutHelp(errorMessage);
+				}
+			});
+		}).future<void>()();
+	}
+
+	private marketplacePlugins: IPlugin[];
+	private getMarketplacePlugins(): IFuture<IPlugin[]> {
+		return ((): IPlugin[] => {
+			if(!this.marketplacePlugins) {
```

Comment:
A better check would be either `!this.marketplacePlugins || !this.marketplacePlugins.length` or just `!this.marketplacePlugins.length`, where should you choose the latter the private field has of course to be declared as `private marketplacePlugins: IPlugin[] = []`.
I am writing this out of concern that should `this.marketplacePlugins` wind up being an empty array then no subsequent attempt will be mate to retrieve the plugins